RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -252,19 +412,26 @@ EvtPlaneProducer::EvtPlaneProducer(const edm::ParameterSet &iConfig)
 
   vertexToken = consumes<std::vector<reco::Vertex>>(vertexTag_);
 
-  caloToken = consumes<CaloTowerCollection>(caloTag_);
-
-  castorToken = consumes<std::vector<reco::CastorTower>>(castorTag_);
-
-  trackToken = consumes<reco::TrackCollection>(trackTag_);
+  strack = trackTag_.label();
+  if (strack.find("packedPFCandidates") != std::string::npos) {
+    packedToken = consumes<edm::View<pat::PackedCandidate>>(trackTag_);
+    lostToken = consumes<edm::View<pat::PackedCandidate>>(losttrackTag_);
```

Comment:
why ```edm::View```, please use the product collection type directly if polymorphic access is not needed